store/src/checkpoints/mod.rs

```diff
@@ -108,7 +108,9 @@ pub trait CheckpointTopology {
 
       let mut balance = self.get_balance(&key)?;
       log::debug!("balance on {check_point_date} {:#?} before operation {balance:?}", op.date);
-      balance += op.to_delta(); // TODO: will fail at inventory operation
+      if op.before != op.after {
```

Comment:
why this check required?

Reply:
For cases when op stays the same but balance changed. Move this in to_delta() again but as first check:
if self.before == self.after {
      BalanceDelta::default()
    }